src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -50,21 +50,25 @@ class FilterButton extends React.Component<Props, State> {
     }
 
     componentDidUpdate(prevProps: Props, prevState: State) {
-        const { columns } = this.props;
-        const { isMenuOpen } = this.state;
+        const { columns, conditions } = this.props;
+        const { isMenuOpen, transientConditions } = this.state;
         const { isMenuOpen: prevIsMenuOpen } = prevState;
         const wasFlyoutOpened = isMenuOpen && !prevIsMenuOpen;
         if (wasFlyoutOpened) {
-            const areConditionsEmpty = this.props.conditions.length === 0;
-            if (areConditionsEmpty) {
-                const transientConditions = columns && columns.length === 0 ? [] : [this.createCondition()];
-                this.setState({
-                    transientConditions,
-                });
-            } else {
-                this.setState({
-                    transientConditions: cloneDeep(this.props.conditions),
-                });
+            const isInitialConditionSet = transientConditions.length > 0;
+            const shouldSetInitialCondition = conditions.length === 0;
+
+            if (!isInitialConditionSet) {
```

Comment:
now that the two conditions are back to back, it's hard to understand what each one signifies.
Can we rename the variables
`isInitialConditionSet` -> `isTransientConditionSet`
`shouldSetInitialCondition` -> `hasDefaultConditions`

We can totally talk through other options, too.